Consul/Client.cs

```diff
@@ -174,8 +174,14 @@ private void ConfigureFromEnvironment(UriBuilder consulAddress)
                 {
                     consulAddress.Host = uri.Host;
                 }
-
-                consulAddress.Port = uri.Port;
+                if (envAddr.Contains($"{uri.Host}:{uri.Port}"))
+                {
+                    consulAddress.Port = uri.Port;
+                }
+                else
+                {
+                    consulAddress.Port = 8500;
+                }
```

Comment:
Assigning the default 8500 here is redundant as the `consulAddress` variable already has the default port pre-assigned. You can remove the `else` block entirely.